api/envoy/data/dns/v3/dns_table.proto

```diff
@@ -39,8 +39,48 @@ message DnsTable {
     }];
   }
 
-  // This message type is extensible and can contain a list of addresses, clusters or
-  // dictate a different method for resolving the addresses for an endpoint
+  // Specify the service protocol using a numeric or string value
+  message DnsServiceProtocol {
+    oneof protocol_config {
+      option (validate.required) = true;
+
+      uint32 number = 1 [(validate.rules).uint32 = {lt: 255}];
+
+      string name = 2 [(validate.rules).string = {min_len: 2 well_known_regex: HTTP_HEADER_NAME}];
+    }
+  }
+
+  // This message defines a service selection record returned for a service query in a domain
+  // [#next-free-field: 8]
+  message DnsService {
+    // The name of the service without the protocol or domain name
+    string service_name = 1;
+
+    // The service protocol. This can be specified as a string or the numeric value of the protocol
+    DnsServiceProtocol protocol = 2;
+
+    // The service entry time to live. This is independent from the DNS Answer record TTL
+    google.protobuf.Duration ttl = 3 [(validate.rules).duration = {gte {seconds: 1}}];
+
+    // The priority of the service record
+    uint32 priority = 4 [(validate.rules).uint32 = {lt: 65536}];
+
+    // The weight of the service record
+    uint32 weight = 5 [(validate.rules).uint32 = {lt: 65536}];
+
+    // The port to which the service is bound
+    uint32 port = 6 [(validate.rules).uint32 = {lt: 65536}];
+
+    // The target address hosting the service
+    string target_address = 7
+        [(validate.rules).string = {min_len: 2 well_known_regex: HTTP_HEADER_NAME}];
```

Comment:
this was the issue that I just fixed.. len(2) req for hostnames is arbitrary

Reply:
I remember that.  I will update.